cmd/connection.go

```diff
@@ -0,0 +1,601 @@
+	api := connectionmanager.New(curl())
+
+	err := api.GrantAccessRoleToConnection(options.connID, options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func connectionAccessRoleRevokeCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "revoke-role-from-connection",
+		Short: "Revoke a permission for a role from a connection",
+		Long:  `Revoke a permission for a role from a connection`,
+		Example: `
+	privx-cli connections revoke-role-from-connection [access flags] --id <CONN-ID> --rid <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionAccessRoleRevoke(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "id", "", "connection ID")
+	flags.StringVar(&options.roleID, "rid", "", "role ID")
+	cmd.MarkFlagRequired("id")
+	cmd.MarkFlagRequired("rid")
+
+	return cmd
+}
+
+func connectionAccessRoleRevoke(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.RevokeAccessRoleFromConnection(options.connID, options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func allConnectionAccessRoleRevokeCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "revoke-role-from-all-connections",
+		Short: "Revoke permissions for a role from connections",
+		Long:  `Revoke permissions for a role from connections`,
+		Example: `
+	privx-cli connections revoke-role-from-all-connections [access flags] --rid <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return allConnectionAccessRoleRevoke(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.roleID, "rid", "", "role ID")
+	cmd.MarkFlagRequired("rid")
+
+	return cmd
+}
+
+func allConnectionAccessRoleRevoke(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.RevokeAccessRoleFromAllConnections(options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func connectionTerminateCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "terminate-connection",
+		Short: "Terminate connection by ID",
+		Long:  `Terminate connection by ID`,
+		Example: `
+	privx-cli connections terminate-connection [access flags] --id <CONN-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionTerminate(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "id", "", "connection ID")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func connectionTerminate(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.TerminateConnection(options.connID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func connectionByTargetHostTerminateCmd() *cobra.Command {
```

Comment:
Again, this is the same action (terminate) but is slightly different from the basic termination. I would put this in a separate function and use flags to change the execution

```
privx-cli connections terminate --conn-id <connection_id>
privx-cli connections terminate --by-target <host_id>
privx-cli connections terminate --by-user <user_id>
```